src/routes/accountOpeningRequest.ts

```diff
@@ -64,14 +69,19 @@ export const createAccountOpeningRequest = async (
     iban: account.iban,
   };
 
-  person.accountOpeningRequests = [
-    ...person.accountOpeningRequests.filter(
-      (request) => request.id !== accountOpeningRequest.id
-    ),
-    completedRequest,
-  ];
-
-  await savePerson(person);
+  await redlock.using([personKey], 5000, async (signal) => {
+    if (signal.aborted) {
+      throw signal.error;
+    }
+    person = await getPerson(personId);
```

Comment:
why account creation is not locked?

Reply:
createAccount function does its own locking and releases the resource when it is done